server/src/test/java/org/jboss/as/server/services/security/VaultHandler.java

```diff
@@ -0,0 +1,377 @@
+                throw new IllegalArgumentException("Wrong keyStoreType. Supported are only (JCEKS or JKS). Preferred is JCEKS.");
+            }
+            this.keyStoreType = keyStoreType;
+        }
+
+        if (keyStorePassword == null) {
+            this.keyStorePassword = DEFAULT_PASSWORD;
+        } else if (keyStorePassword.startsWith(PicketBoxSecurityVault.PASS_MASK_PREFIX)) {
+            throw new IllegalArgumentException("keyStorePassword cannot be a masked password, use plain text password, please");
+        } else {
+            this.keyStorePassword = keyStorePassword;
+        }
+
+        try {
+            File keyStoreFile = new File(keyStore);
+            if (!keyStoreFile.exists()) {
+                if (!this.keyStoreType.equals("JCEKS")) {
+                    throw new RuntimeException("keyStoreType has to be JCEKS when creating new key store");
+                }
+                File keyStoreParent = keyStoreFile.getAbsoluteFile().getParentFile();
+                if (keyStoreParent != null) {
+                    if (!keyStoreParent.exists()) {
+                        assert keyStoreParent.mkdirs();
+                    } else {
+                        assert keyStoreParent.isDirectory();
+                    }
+                }
+                KeyStore ks = KeyStoreUtil.createKeyStore(this.keyStoreType, this.keyStorePassword.toCharArray());
+                KeyGenerator keyGenerator = KeyGenerator.getInstance("AES");
+                keyGenerator.init(this.keySize);
+                SecretKey secretKey = keyGenerator.generateKey();
+                KeyStore.SecretKeyEntry skEntry = new KeyStore.SecretKeyEntry(secretKey);
+                KeyStore.PasswordProtection p = new KeyStore.PasswordProtection(this.keyStorePassword.toCharArray());
+                ks.setEntry(this.alias, skEntry, p);
+                ks.store(new FileOutputStream(keyStoreFile), this.keyStorePassword.toCharArray());
+            }
+            this.keyStore = keyStoreFile.getAbsolutePath();
+        } catch (Exception e) {
+            throw new RuntimeException("Problem creating keyStore: ", e);
+        }
+
+        File vaultDirectory = new File(encodedVaultFileDirectory).getAbsoluteFile();
+
+        if (!vaultDirectory.exists()) {
+            assert vaultDirectory.mkdirs();
+        } else if (!vaultDirectory.isDirectory()) {
+            throw new RuntimeException("Vault encryption directory has to be directory, but "
+                    + vaultDirectory.getAbsolutePath() + " is not.");
+        }
+
+        this.encodedVaultFileDirectory = vaultDirectory.getAbsolutePath();
+
+        if (salt == null) {
+            String tmp = Long.toHexString(System.currentTimeMillis())
+                    + Long.toHexString(System.currentTimeMillis())
+                    + Long.toHexString(System.currentTimeMillis())
+                    + Long.toHexString(System.currentTimeMillis());
+            this.salt = tmp.substring(0, 8);
```

Comment:
I'm not sure the concatenation is needed if we're just using the first 8 characters of the string. The first `Long.toHexString(System.currentTimeMillis())` will always generate a hex string longer than 8 characters.